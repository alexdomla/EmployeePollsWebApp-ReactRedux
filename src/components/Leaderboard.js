import React from "react";
import { connect } from "react-redux";
import Nav from "./Nav";

// This component displays the leaderboard with user scores.
const Leaderboard = ({ authedUser, users }) => {
  return (
    <div className="leaderboardPage">
      {/* If the user is authenticated, show the navigation bar */}
      {authedUser ? <Nav></Nav> : null}
      <h1>Leaderboard Page</h1>
      <table className="">
        <thead className="">
          <tr className="">
            <th className="">Avatar</th>
            <th className="">Name</th>
            <th className="">Answered</th>
            <th className="">Created</th>
          </tr>
        </thead>
        <tbody className="">
          {/* Iterate over the users array and display each user's information */}
          {users.map((user) => (
            <tr key={user.id}>
              <td>
                <div className="table__img">
                  {/* Display the user's avatar */}
                  <img src={user.avatarURL} alt="" />
                </div>
              </td>
              <td className="">
                {/* Display the user's name */}
                <span className="">{user.name}</span>
              </td>
              <td className="">{Object.keys(user.answers).length}</td>
              {/* Display the number of questions the user has asked */}
              <td className="">{user.questions.length}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

// This function maps the state variables to the component props.
const mapStateToProps = ({ authedUser, users }) => ({
  authedUser,
  // Sort the users array in descending order of number of answers to questions.
  users: Object.values(users).sort(
    (a, b) => Object.keys(b.answers).length - Object.keys(a.answers).length
  ),
});

// Connect the component to the Redux store.
export default connect(mapStateToProps)(Leaderboard);
