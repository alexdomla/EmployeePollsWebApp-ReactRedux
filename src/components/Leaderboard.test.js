import { store } from "../store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Leaderboard from "../components/Leaderboard";

describe("leaderboard", () => {
  it("will match snapshot", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Leaderboard />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
