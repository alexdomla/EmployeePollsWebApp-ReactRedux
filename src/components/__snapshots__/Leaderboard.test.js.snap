// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leaderboard will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaderboardPage"
      >
        <h1>
          Leaderboard Page
        </h1>
        <table
          class=""
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                Avatar
              </th>
              <th
                class=""
              >
                Name
              </th>
              <th
                class=""
              >
                Answered
              </th>
              <th
                class=""
              >
                Created
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaderboardPage"
    >
      <h1>
        Leaderboard Page
      </h1>
      <table
        class=""
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class=""
            >
              Avatar
            </th>
            <th
              class=""
            >
              Name
            </th>
            <th
              class=""
            >
              Answered
            </th>
            <th
              class=""
            >
              Created
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
